vmdb/spec/requests/api/vms_spec.rb

```diff
@@ -2,20 +2,30 @@
 # REST API Request Tests - /api/vms
 #
 require 'spec_helper'
+require 'requests/shared_examples/api'
 
 describe ApiController do
 
   include Rack::Test::Methods
 
-  before(:each) do
-    init_api_spec_env
+  let(:zone)       { FactoryGirl.create(:zone, :name => "api_zone") }
+  let(:miq_server) { FactoryGirl.create(:miq_server, :guid => miq_server_guid, :zone => zone) }
+  let(:ems)        { FactoryGirl.create(:ems_vmware, :zone => zone) }
+  let(:host)       { FactoryGirl.create(:host) }
+
+  let(:vm)         { FactoryGirl.create(:vm_vmware, :host => host, :ems_id => ems.id, :raw_power_state => "poweredOn") }
+  let(:vm1)        { FactoryGirl.create(:vm_vmware, :host => host, :ems_id => ems.id, :raw_power_state => "poweredOn") }
+  let(:vm2)        { FactoryGirl.create(:vm_vmware, :host => host, :ems_id => ems.id, :raw_power_state => "poweredOn") }
+  let(:vm1_url)    { vm_url(vm1.id) }
+  let(:vm2_url)    { vm_url(vm2.id) }
+  let(:vms_list)   { [vm1_url, vm2_url] }
+  let(:vm_guid)    { vm.guid }
+  let(:vm_href)    { vm_url(vm.id) }
 
-    @zone       = FactoryGirl.create(:zone, :name => "api_zone")
-    @miq_server = FactoryGirl.create(:miq_server, :guid => miq_server_guid, :zone => @zone)
-    @ems        = FactoryGirl.create(:ems_vmware, :zone => @zone)
-    @host       = FactoryGirl.create(:host)
+  let(:invalid_vm_url) { vm_url(999_999) }
 
-    Host.any_instance.stub(:miq_proxy).and_return(@miq_server)
+  before(:each) do
+    init_api_spec_env
```

Comment:
remote_guid_miq_server_zone method is already called in the the init_api_sec_env method.